src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs

```diff
@@ -50,6 +51,18 @@ public AdvancedOptionPageControl(OptionStore optionStore, IComponentModel compon
 
             InitializeComponent();
 
+            BindToOptions();
+        }
+
+        public AdvancedOptionPageControl(OptionStore optionStore) : base(optionStore)
+        {
+            InitializeComponent();
+
+            BindToOptions();
+        }
```

Comment:
I can mark this as `[Obsolete("Used for tests only")]` and suppress the obsoletion warning in test. But I don't think this is the best approach to go. I'm using it for now since it's what I could get working :)

Tagging @sharwell for help writing the test in a better way :)

Reply:
I'd remove `IComponentModel componentModel` parameter from the above ctor and instead take `IThreadingContext` and `ColorSchemeApplier` directly as parameters. These can then be mocked by the tests. Might need to introduce interface `IColorSchemeApplier` to allow such mocking.

Reply:
@tmat Instead of doing that, I'd like to instantiate `AdvancedOptionPage` instead of `AdvancedOptionPageControl` in tests. Is that possible?

That would avoid the extra constructor, avoid instantiating `SolutionOptionSet`, and avoid calling `ILegacyWorkspaceOptionService.SetOptions`

I tried to do that but it didn't work. I sent you an email with details.